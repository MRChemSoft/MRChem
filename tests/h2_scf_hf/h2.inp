# vim:syntax=sh:

world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}

Basis {
  order = 7                       # Polynomial order
  type = Legendre                 # Polynomial type (Legendre or Interpolating)
}


Molecule {
$coords
H   0.0     0.0    -0.7
H   0.0     0.0     0.7
$end
}

WaveFunction {
  method = HF                     # Wave function method (HF or DFT)
}

Properties {
  dipole_moment = true            # Compute ground state energy
}

SCF {
  kain = 3                        # Length of KAIN iterative history
  max_iter = 5
  orbital_thrs = 1.0e-2           # Convergence threshold in orbital residual
  guess_type = SAD_DZ             # Type of initial guess: none, mw, gto
}

