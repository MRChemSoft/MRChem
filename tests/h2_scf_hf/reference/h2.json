{
  "input": {
    "molecule": {
      "charge": 0,
      "coords": [
        {
          "atom": "h",
          "xyz": [
            0.0,
            0.0,
            -0.7
          ]
        },
        {
          "atom": "h",
          "xyz": [
            0.0,
            0.0,
            0.7
          ]
        }
      ],
      "multiplicity": 1
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 7,
      "basis_type": "legendre",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "h2.inp",
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        }
      },
      "initial_guess": {
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "Hartree-Fock",
        "prec": 0.001,
        "restricted": true,
        "type": "sad",
        "zeta": 2
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "energy_thrs": -1.0,
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 3,
        "localize": false,
        "max_iter": 5,
        "method": "Hartree-Fock",
        "orbital_thrs": 0.01,
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        -1.1189687543466913e-17
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 0.00025324473011903415,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            0.0,
            0.0,
            0.00025324473011903415
          ],
          "vector_el": [
            0.0,
            0.0,
            0.00025324473012147664
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.0
          ]
        }
      },
      "geometry": [
        {
          "symbol": "H",
          "xyz": [
            0.0,
            0.0,
            -0.7
          ]
        },
        {
          "symbol": "H",
          "xyz": [
            0.0,
            0.0,
            0.7
          ]
        }
      ],
      "multiplicity": 1,
      "orbital_energies": {
        "energy": [
          -0.5959952239811064
        ],
        "occupation": [
          2.0
        ],
        "spin": [
          "p"
        ],
        "sum_occupied": -1.1919904479622128
      },
      "scf_energy": {
        "E_ee": 1.3121427165845492,
        "E_eext": 0.0,
        "E_el": -1.848064491938466,
        "E_en": -3.621006751204245,
        "E_kin": 1.1168682152895177,
        "E_next": 0.0,
        "E_nn": 0.7142857142857143,
        "E_nuc": 0.7142857142857143,
        "E_tot": -1.1337787776527515,
        "E_x": -0.656068672608288,
        "E_xc": 0.0
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 1,
      "routine": "mrchem.x",
      "total_cores": 1,
      "version": "1.1.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 1.1787994282038419,
        "E_eext": 0.0,
        "E_el": -1.8126340620994028,
        "E_en": -3.2892103512934012,
        "E_kin": 0.8871016495907046,
        "E_next": 0.0,
        "E_nn": 0.7142857142857143,
        "E_nuc": 0.7142857142857143,
        "E_tot": -1.0983483478136886,
        "E_x": -0.5893247886005482,
        "E_xc": 0.0
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 1.2655515726074755,
              "E_eext": 0.0,
              "E_el": -1.8437367303811687,
              "E_en": -3.505803088720751,
              "E_kin": 1.0292875318778836,
              "E_next": 0.0,
              "E_nn": 0.7142857142857143,
              "E_nuc": 0.7142857142857143,
              "E_tot": -1.1294510160954543,
              "E_x": -0.6327727461457768,
              "E_xc": 0.0
            },
            "energy_total": -1.1294510160954543,
            "energy_update": 0.031102668281765666,
            "mo_residual": 0.0808962276271831,
            "wall_time": 4.861896886
          },
          {
            "energy_terms": {
              "E_ee": 1.3004666531488585,
              "E_eext": 0.0,
              "E_el": -1.8476537909256674,
              "E_en": -3.591947339814375,
              "E_kin": 1.0940574474938676,
              "E_next": 0.0,
              "E_nn": 0.7142857142857143,
              "E_nuc": 0.7142857142857143,
              "E_tot": -1.133368076639953,
              "E_x": -0.6502305517540184,
              "E_xc": 0.0
            },
            "energy_total": -1.133368076639953,
            "energy_update": 0.003917060544498696,
            "mo_residual": 0.02755835055766473,
            "wall_time": 4.775336529
          },
          {
            "energy_terms": {
              "E_ee": 1.3121427165845492,
              "E_eext": 0.0,
              "E_el": -1.848064491938466,
              "E_en": -3.621006751204245,
              "E_kin": 1.1168682152895177,
              "E_next": 0.0,
              "E_nn": 0.7142857142857143,
              "E_nuc": 0.7142857142857143,
              "E_tot": -1.1337787776527515,
              "E_x": -0.656068672608288,
              "E_xc": 0.0
            },
            "energy_total": -1.1337787776527515,
            "energy_update": 0.0004107010127984978,
            "mo_residual": 0.008740657809113301,
            "wall_time": 4.747623383
          }
        ],
        "wall_time": 14.384979977
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
