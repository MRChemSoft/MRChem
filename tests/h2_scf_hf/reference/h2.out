                                                                      
                                                                      
**********************************************************************
***                                                                ***
***                                                                ***
***        __  __ ____   ____ _                                    ***
***       |  \/  |  _ \ / ___| |__   ___ _ __ ___                  ***
***       | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                 ***
***       | |  | |  _ <| |___| | | |  __/ | | | | |                ***
***       |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                ***
***                                                                ***
***       VERSION            0.3.0-alpha                           ***
***                                                                ***
***       Git branch         modernize_energy_calculation          ***
***       Git commit hash    f869d887b25c726c881d-dirty            ***
***       Git commit author  Stig Rune Jensen                      ***
***       Git commit date    Thu Mar 5 12:03:21 2020 +0100         ***
***                                                                ***
***       Magnar Bjorgve     <magnar.bjorgve@uit.no>               ***
***       Roberto Di Remigio <roberto.di.remigio@uit.no>           ***
***       Luca Frediani      <luca.frediani@uit.no>                ***
***       Gabriel Gerez      <gsa017@post.uit.no>                  ***
***       Stig Rune Jensen   <stig.r.jensen@uit.no>                ***
***       Peter Wind         <peter.wind@uit.no>                   ***
***                                                                ***
**********************************************************************

----------------------------------------------------------------------

 MPI processes         :      (no bank)                             1
 OpenMP threads        :                                            1
 Total cores           :                                            1
                                                                      
----------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

----------------------------------------------------------------------

 MRCPP version         : 1.2.0-alpha2
 Git branch            : modernize_energy_calculation
 Git commit hash       : ea07b65c8109b8937518-dirty
 Git commit author     : Magnar Bj√∏rgve
 Git commit date       : Wed Mar 4 22:18:16 2020 +0100

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : OpenMP

----------------------------------------------------------------------


                                                                      
======================================================================
                       MultiResolution Analysis
----------------------------------------------------------------------
 polynomial order      : 7
 polynomial type       : Legendre
----------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***                     Initializing Molecule                      ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
======================================================================
                               Molecule
----------------------------------------------------------------------
 Charge                :                                            0
 Multiplicity          :                                            1
----------------------------------------------------------------------
    N    Atom          :              x              y              z
----------------------------------------------------------------------
    0       H          :       0.000000       0.000000      -0.700000
    1       H          :       0.000000       0.000000       0.700000
----------------------------------------------------------------------
 Center of mass        :       0.000000       0.000000      -0.000000
 Gauge origin          :       0.000000       0.000000       0.000000
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***              Computing Initial Guess Wavefunction              ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Diagonalize Hamiltonian matrix
 Precision             : 1.00000e-03
 Restricted            : True
 Hamiltonian           : Superposition of Atomic Densities (SAD)
 Functional            : LDA (SVWN5)
 AO basis              : Hydrogenic orbitals
 Zeta quality          : 2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
                          Molecular Orbitals
----------------------------------------------------------------------
 Alpha electrons       :                                            1
 Beta electrons        :                                            1
 Total electrons       :                                            2
----------------------------------------------------------------------
    n  Occ Spin        :                                         Norm
----------------------------------------------------------------------
    0    2    p        :                           9.999998867959e-01
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***              Computing Ground State Wavefunction               ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Compute initial energy
 Method                : Hartree-Fock
 Precision             : 1.00000e-03
 Localization          : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
                           Molecular Energy
----------------------------------------------------------------------
 Kinetic energy        :           (au)                0.887101649591
 E-N energy            :           (au)               -3.289210351293
 Coulomb energy        :           (au)                1.178799428204
 Exchange energy       :           (au)               -0.589401251813
 X-C energy            :           (au)                0.000000000000
 Ext. field (el)       :           (au)                0.000000000000
 Ext. field (nuc)      :           (au)                0.000000000000
----------------------------------------------------------------------
 Electronic energy     :           (au)               -1.812710525312
 Nuclear energy        :           (au)                0.714285714286
----------------------------------------------------------------------
 Total energy          :           (au)           -1.098424811026e+00
                       :     (kcal/mol)           -6.892719699033e+02
                       :       (kJ/mol)           -2.883913923947e+03
                       :           (eV)           -2.988966048130e+01
======================================================================


======================================================================
                           Orbital Energies
----------------------------------------------------------------------
    n  Occ Spin        :                                      Epsilon
----------------------------------------------------------------------
    0    2    p        :           (au)               -0.611656312944
----------------------------------------------------------------------
 Sum occupied          :           (au)               -1.223312625889
======================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Optimize ground state orbitals
 Method                : Hartree-Fock
 Max iterations        : 5
 KAIN solver           : 3
 Localization          : Off
 Diagonalization       : First two iterations
 Helmholtz precision   : 1.00000e-03
 Start precision       : 1.00000e-03
 Final precision       : 1.00000e-03
 Energy threshold      : Off
 Orbital threshold     : 1.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
 Iter         MO residual            Total energy              Update
----------------------------------------------------------------------
    0        1.000000e+00         -1.098424811026       -1.098425e+00
    1        8.091939e-02         -1.129611786251       -3.118698e-02
    2        2.765730e-02         -1.133576810812       -3.965025e-03
    3        8.810274e-03         -1.133996586306       -4.197755e-04
----------------------------------------------------------------------
                   SCF converged in 3 iterations!
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***                 Printing Molecular Properties                  ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
======================================================================
                               Molecule
----------------------------------------------------------------------
 Charge                :                                            0
 Multiplicity          :                                            1
----------------------------------------------------------------------
    N    Atom          :              x              y              z
----------------------------------------------------------------------
    0       H          :       0.000000       0.000000      -0.700000
    1       H          :       0.000000       0.000000       0.700000
----------------------------------------------------------------------
 Center of mass        :       0.000000       0.000000      -0.000000
 Gauge origin          :       0.000000       0.000000       0.000000
======================================================================


======================================================================
                           Orbital Energies
----------------------------------------------------------------------
    n  Occ Spin        :                                      Epsilon
----------------------------------------------------------------------
    0    2    p        :           (au)               -0.596133571590
----------------------------------------------------------------------
 Sum occupied          :           (au)               -1.192267143179
======================================================================


======================================================================
                           Molecular Energy
----------------------------------------------------------------------
 Kinetic energy        :           (au)                1.117767387200
 E-N energy            :           (au)               -3.622064845205
 Coulomb energy        :           (au)                1.312451048159
 Exchange energy       :           (au)               -0.656435890746
 X-C energy            :           (au)                0.000000000000
 Ext. field (el)       :           (au)                0.000000000000
 Ext. field (nuc)      :           (au)                0.000000000000
----------------------------------------------------------------------
 Electronic energy     :           (au)               -1.848282300592
 Nuclear energy        :           (au)                0.714285714286
----------------------------------------------------------------------
 Total energy          :           (au)           -1.133996586306e+00
                       :     (kcal/mol)           -7.115935957208e+02
                       :       (kJ/mol)           -2.977307606428e+03
                       :           (eV)           -3.085761775536e+01
======================================================================


======================================================================
                            Dipole Moment
----------------------------------------------------------------------
 Electronic vector     :      -0.000000      -0.000000       0.000254
 Magnitude             :           (au)                      0.000254
                       :        (Debye)                      0.000645
----------------------------------------------------------------------
 Nuclear vector        :       0.000000       0.000000      -0.000000
 Magnitude             :           (au)                      0.000000
                       :        (Debye)                      0.000000
----------------------------------------------------------------------
 Total vector          :       0.000000       0.000000       0.000254
 Magnitude             :           (au)                      0.000254
                       :        (Debye)                      0.000645
======================================================================


                                                                      
                                                                      
**********************************************************************
***                                                                ***
***                         Exiting MRChem                         ***
***                                                                ***
***                    Wall time :  0h  0m 23s                     ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
